catalog/plugin/catalog-plugin-async/src/main/java/org/codice/ddf/catalog/plugin/async/api/internal/ProcessResource.java

```diff
@@ -0,0 +1,101 @@
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.plugin.async.api.internal;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ *
+ * A {@code ProcessResource} represents a local or remote resource that will be used during processing
+ * by the {@link ProcessingFramework}.
+ */
+public interface ProcessResource {
+
+    String CONTENT_SCHEME = "content";
+
+    String DEFAULT_MIME_TYPE = "application/octet-stream";
+
+    String DEFAULT_FILE_NAME = "content_store_file.bin";
+
+    String QUALIFIER = "qualifier";
+
+    /**
+     * Return the globally unique ID for the {@link ProcessResource}.
+     *
+     * @return the {@link ProcessResource}'s ID
+     */
+    String getId();
+
+    /**
+     * Return the URI of the {@link ProcessResource}.
+     *
+     * @return the URI of the {@link ProcessResource}
+     */
+    String getUri();
+
+    /**
+     * An optional field to provide additional information about a {@link ProcessResource}
+     *
+     * @return the qualifier of the {@link ProcessResource}
+     */
+    String getQualifier();
+
+    /**
+     * The filename of the {@link ProcessResource}. If not set it will default to {@link #DEFAULT_FILE_NAME}
+     *
+     * @return the filename of the {@link ProcessResource}
+     */
+    String getFilename();
+
+    /**
+     * Return the mime type raw data for the {@link ProcessResource}, e.g., image/nitf or application/json;id=geojson
+     *
+     * @return the mime type raw data for the {@link ProcessResource}
+     */
+    String getMimeTypeRawData();
+
+    /**
+     * Return the input stream containing the {@link ProcessResource}'s actual data content.
+     *
+     * @return the {@link ProcessResource}'s input stream
+     * @throws IOException if the input stream is not available
+     */
+    InputStream getInputStream() throws IOException;
+
+    /**
+     * Return the total number of bytes in the {@link ProcessResource}'s input stream.
+     *
+     * @return returns the total number of bytes that can be read from the input stream
+     * @throws IOException if the total number of bytes is unknown or not applicable (e.g., in the case of a
+     *                     live stream)
+     */
+    long getSize() throws IOException;
+
+    /**
+     * Determines if the {@code ProcessResource} has been modified during processing.
+     *
+     * @return {@code true} if modified, {@code false} otherwise
+     */
+    boolean isModified();
+
+    /**
+     * Marks the {@code ProcessResource} as modified.
+     */
+    void markAsModified();
```

Comment:
The purpose of this and other like methods was that, since the `PostProcessPlugin`s may not touch or modify anything during their processing, we needed a way to show that. This allowed us not to make any calls back to the `CatalogFramework`, such as new creates or updates. Our specific example was, even though we had one `PostProcessPlugin` that touched only NITFs products, we were sending erroneous requests back to the `Catalog` for products like a JPG.